wrappers/src/query_cs.cpp

```diff
@@ -391,4 +392,38 @@ REALM_EXPORT Results* query_create_sorted_results(Query * query_ptr, SharedRealm
     });
 }
 
+
+#pragma mark  PrimaryKey Searches
+Row* row_for_primarykey(Table* table_ptr, ObjectSchema* object_schema, std::function<size_t(Table*, size_t columnIndex)> finder, NativeException::Marshallable& ex)
+{
+    return handle_errors(ex, [&]() {
+        size_t columnIndex = realm::npos;
+        if (object_schema->primary_key.size() > 0)
+            columnIndex = table_ptr->get_column_index(object_schema->primary_key);
+        if (columnIndex == realm::npos)
+            throw PrimaryKeyNotDeclaredException(object_schema->name);
```

Comment:
The motivation of my suggestion is to use the object schema's cached version of the column index rather than retrieving it from the table.

It's programmer error to try to look up an object by primary key when the class has no primary key defined. I don't see why it's useful to optimize for that case?